screensharing/render.js

```diff
@@ -0,0 +1,164 @@
+/* global process */
+
+const { ipcRenderer, desktopCapturer } = require('electron');
+const semver = require('semver');
+
+const { SCREEN_SHARE_EVENTS_CHANNEL, SCREEN_SHARE_EVENTS } = require('./constants');
+
+/**
+ * Renderer process component that sets up electron specific screen sharing functionality, like screen sharing
+ * marker and window selection.
+ * {@link ScreenShareMainHook} needs to be initialized in the main process for the always on top tracker window
+ * to work.
+ */
+class ScreenShareRenderHook {
+    /**
+     * Creates a ScreenShareRenderHook hooked to jitsi meet iframe events.
+     *
+     * @param {iframe} iframe - The jitsi Meet iframe object. This may seem redundant as we can access it from
+     * the api object, however we need it here for backwards compatibility as some clients are setup is such a
+     * way that they may end up calling the old jitsi-meet-electron-utils with the updated api, causing crashes.
+     * @param {JitsiIFrameApi} api - The Jitsi Meet iframe api object.
+     */
+    constructor(iframe, api) {
```

Comment:
`iframe` is kind of redundant since you can get it from `api`

Reply:
It is, i left a comment in the parameter section explaining why it's necessary. Basically jitsi-meet-electron-utils is linked to 8x8-meet-electron (standalone) and VOD electron apps, however the code calling it is deployed separately (8x8.vc) and loaded in electron via (iframe), which means that once we make the changes to 8x8.vc to call `new ScreenShareMainHook(api)` instead of new `ScreenShareMainHook(iframe)`  all the old apps (standalone and vod) will load these new changes in the frame which will result in the iframe calling older versions of jtisi-meet-electron-utils inadvertently breaking it.
Long story short we can't change the api.

Reply:
OK. I see. I'm sorry I missed the comment. Let me thing about it and discuss offline.

Reply:
OK. I see. I'm sorry I missed the comment. Let me thing about it and discuss offline.